core/eval.go

```diff
@@ -1464,6 +1464,117 @@ func evalBITOP(args []string) []byte {
 	}
 }
 
+// BITPOS key bit [start [end [BYTE | BIT]]]
+func evalBITPOS(args []string) []byte {
+	// if more than 5 arguments are provided or less than 2 return error
+	if len(args) > 5 || len(args) < 2 {
+		return Encode(errors.New("ERR syntax error"), false)
+	}
+
+	// appending args with constant values
+	if len(args) == 2 {
+		args = append(args, "0")
+	}
+	if len(args) == 3 {
+		args = append(args, "-1")
+	}
+	if len(args) == 4 {
+		args = append(args, "BYTE")
+	}
```

Comment:
@soumya-codes if you want this can be done by storing the variable of args length in variable and then applying switch case, to prevent the length check again and again.

I went ahead with this approach because it felt more cleaner 😅 
Do let me know your thoughts on this!

Reply:
@MohitVachhani I tried my hand to see if we can simplify the code. Let me know what you think about the implementation [here](https://gist.github.com/soumya-codes/29fec9697a597ab3f5b23e0f34678697).